### PR TITLE
Merge gob branch with master before collecting functions.

### PR DESCRIPTION
**Pull request by etefera**
Fixes https://github.com/GoogleContainerTools/kpt/issues/2446

### CODE REVIEW

File: .github/workflows/firebase-hosting-merge.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ jobs:
       - uses: actions/checkout@v2
```
Comment (mengqiy) [inline .github/workflows/firebase-hosting-merge.yml:14]: We should add `gob` as a trigger branches, right?

File: .github/workflows/checkSite.yml [hunk 2]
```diff
@@ -27,6 +27,8 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
+      - name: Merge gob branch
+        run: git --work-tree=. checkout gob .
```
Comment (mengqiy) [inline .github/workflows/checkSite.yml:31]: This command doesn't work. It seems the following works.
```
git worktree add ./gob gob
cp -r gob/* .
```

File: .github/workflows/firebase-hosting-merge.yml (hunk 1 above)
Comment (mengqiy) [reply .github/workflows/firebase-hosting-merge.yml:14]: Add it to checkSite.yml

File: .github/workflows/checkSite.yml (hunk 2 above)
Comment (etefera) [reply .github/workflows/checkSite.yml]: This command worked with `upstream/gob` in place of `gob`. Switched it to your command.

File: .github/workflows/firebase-hosting-merge.yml (hunk 1 above)
Comment (etefera) [reply .github/workflows/firebase-hosting-merge.yml:15]: Done. checkSite is triggered on every branch.

**Review by mengqiy — APPROVED** [review]

**Pull request merged**